pysatl_cpd/cpd_solver.py

```diff
@@ -0,0 +1,186 @@
+            diff = ";".join(map(str, self.result_diff))
+            method_output += f"Expected change point: ({expected_cp_results})\n"
+            method_output += f"Difference: ({diff})\n"
+        method_output += f"Computation time (sec): {round(self.time_sec, 2)}"
+        return method_output
+
+    def count_confusion_matrix(self, window: tuple[int, int] | None = None) -> tuple[int, int, int, int]:
+        """method for counting confusion matrix for hypothesis of equality of CPD results and expected
+        results on a window
+
+        :param: window: tuple of two indices (start, stop), determines a window for hypothesis
+
+        :return: tuple of integers (true-positive, true-negative, false-positive, false-negative)
+        """
+        if self.expected_result is None:
+            raise ValueError("this object is not provided with expected result, confusion matrix cannot be calculated")
+        return CpdResultsAnalyzer.count_confusion_matrix(self.result, self.expected_result, window)
+
+    def count_accuracy(self, window: tuple[int, int] | None = None) -> float:
+        """method for counting accuracy metric for hypothesis of equality of CPD results and expected
+        results on a window
+
+        :param: window: tuple of two indices (start, stop), determines a window for hypothesis
+
+        :return: float, accuracy metric
+        """
+        if self.expected_result is None:
+            raise ValueError("this object is not provided with expected result, accuracy cannot be calculated")
+        return CpdResultsAnalyzer.count_accuracy(self.result, self.expected_result, window)
+
+    def count_precision(self, window: tuple[int, int] | None = None) -> float:
+        """method for counting precision metric for hypothesis of equality of CPD results and expected
+        results on a window
+
+        :param: window: tuple of two indices (start, stop), determines a window for hypothesis
+
+        :return: float, precision metric
+        """
+        if self.expected_result is None:
+            raise ValueError("this object is not provided with expected result, precision cannot be calculated")
+        return CpdResultsAnalyzer.count_precision(self.result, self.expected_result, window)
+
+    def count_recall(self, window: tuple[int, int] | None = None) -> float:
+        """method for counting recall metric for hypothesis of equality of CPD results and expected results on a window
+
+        :param: window: tuple of two indices (start, stop), determines a window for hypothesis
+
+        :return: float, recall metric
+        """
+        if self.expected_result is None:
+            raise ValueError("this object is not provided with expected result, recall cannot be calculated")
+        return CpdResultsAnalyzer.count_recall(self.result, self.expected_result, window)
+
+    def visualize(self, to_show: bool = True, output_directory: Path | None = None, name: str = "Graph") -> None:
```

Comment:
Maybe "plot" instead of "graph" so we don't have two entities with a same name (graph-based algorithm)? Also related to the doc-string.

Reply:
Done